packages/pieces/src/framework/property/oauth-prop.ts

```diff
@@ -0,0 +1,21 @@
+import { BasePropertySchema, PropertyType, TPropertyValue } from "./base-prop";
+import { AuthProp } from "./custom-auth-prop";
+
+type OAuthUrlResolver = (propsValue: Record<string, AuthProp>) => Promise<string>;
```

Comment:
The counter use case for this is salesforces.

Salesforces requires a static dropdown where you select sandbox or production and depend on that calculate the url value, I guess this should be any prop value.